### ISSUE TITLE
Stake not marked as recovered (awaiting Runtime upgrade)

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/pioneer#3143]
**Taken at**: https://dao.joystream.org/#/election/past-votes
**Fields**:
- Houston, we have a problem.: Stake not marked as recovered
- Comment: My steak is not marked as restored. I have already restored it and used the funds again. Sometimes I work from two devices.

![SCREENSHOT](https://resources.usersnap.com/company/f9f719df-a929-4461-bb6e-46dbfbceffb7/datapoint_screenshot/c7ed1209-fd82-4075-9bd3-fe77b5270c7b-annotated_6bfe6442-4b6c-4c2f-a088-c7f9c274f96b.png?etag=32ced7063064a4c8a5d9afa4da85b505)
*[Download original image](https://resources.usersnap.com/company/f9f719df-a929-4461-bb6e-46dbfbceffb7/datapoint_screenshot/c7ed1209-fd82-4075-9bd3-fe77b5270c7b-annotated_6bfe6442-4b6c-4c2f-a088-c7f9c274f96b.png?etag=32ced7063064a4c8a5d9afa4da85b505)*

**Browser**: Chrome 102 (Windows 10)
**Screen size**: 1920x1200
**Browser size**: 1920x1057
**[View screen recording](https://resources.usersnap.com/company/f9f719df-a929-4461-bb6e-46dbfbceffb7/screen_recording/de06fdcc-0a99-4831-a437-76c9dc1996dd_screen_recording_4c90a23e-e01c-40d3-aa4f-128281c9b169.webm?etag=23594c12160a47502d6f0f7d7719af60)**
**[Open #268 in Usersnap](https://app.usersnap.com/l/feedback/d25e90ad-8481-4387-9871-b2ae8ee6b512?utm_medium=integration&utm_campaign=integration&utm_content=open)**

Powered by **[Usersnap](https://usersnap.com/?utm_source=product&amp;utm_medium=poweredbylink&amp;utm_campaign=github_entry)**.

**Comment by mochet** [comment]
Another user reported the same problem on Discord.

The previous votes section contains several votes which no longer exist within `chain state` which meant the user thought that they had 2m tJOY locked--in truth the amount was only 10k tjoy
![image](https://user-images.githubusercontent.com/50830161/171466046-f8d8e55c-971c-4857-8a4a-1152cb1b33d0.png)

![image](https://user-images.githubusercontent.com/50830161/171466226-57917e13-4a66-48bc-8d2f-820ca00843ee.png)
![image](https://user-images.githubusercontent.com/50830161/171466254-123ffe1a-c53e-4892-bffc-15d09a208679.png)

**Comment by dmtrjsg** [comment]
@rafalpawlow @thesan do we store the state of stake recovered in local storage?  So when accessed from a diff device/ browser the state is not shown as up to date?

**Comment by ivanturlakov** [comment]
Tested on staging.

The situation is similar to that described by Dmitry in the first post. But lately, I have been using only one device

No recoverable locks in account section
<img width="1176" alt="Снимок экрана 2022-06-14 в 12 49 50" src="https://user-images.githubusercontent.com/46903215/173551485-8f06b6f2-e562-4a64-a97e-7ad6c2a14410.png">

But one of my past votes shows as recoverable
<img width="1178" alt="Снимок экрана 2022-06-14 в 13 05 05" src="https://user-images.githubusercontent.com/46903215/173551968-3d857776-4d39-4a96-ad2e-30aa7695c79e.png">

Error if I try to recover
<img width="1425" alt="Снимок экрана 2022-06-14 в 11 50 52" src="https://user-images.githubusercontent.com/46903215/173552883-fa668eca-943e-4f72-ac83-0487ff466be1.png">
more details at: [polkadot](https://polkadot.js.org/apps/?rpc=wss://3.235.196.1.nip.io/ws-rpc#/explorer/query/0x2bf801793ec01a646fa90652dbed665911e08d3b8ab9184ecb75c654ff5e093a)

**Comment by dmtrjsg** [comment]
⚠️  It's a runtime issue. Moving this to On Hold/ Blocked.

**Comment by dmtrjsg** [comment]
seem to be unblocked now as linked issue is closed

**Comment by dmtrjsg** [comment]
Seems to be already resolved in 
- https://github.com/Joystream/pioneer/issues/3303 

@polikosi can you confirm if you have managed to successfully recover voting stake on Carthage?

**Comment by polikosi** [comment]
@dmtrjsg yes, I confirm that everything was successful for me.
Just tried again - https://polkadot.js.org/apps/?rpc=wss://54.172.154.45.nip.io/ws-rpc#/explorer/query/0xf4b7e7e1c7654f13c98694dffb2df38525f99fda4040a33b164e7542980036ad

**Comment by dmtrjsg** [comment]
@polikosi great, thanks for confirming 🙌
